pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
 
 __all__ = [
     'CHTest',
+    'decompose',
     'OCSBTest'
 ]
 
 
+def decompose(x, type, f, filter=None):
+    """
+
+    Parameters
+    ----------
+    x : np.array
+        The time series of which the trend, seasonal, and noise/random
+        components will be extracted.
+
+    type: str
```

Comment:
Do you think it's worth starting with a default? Or is that something we should not make assumptions about? Also, how do you feel about renaming this to `typ` since `type` is a reserved word in Python?

Reply:
I would not put a default. I would force the user to be mindful about what is happening. I believe R does the same thing. They do filter the acceptable values to `multiplicative` or `additive`. See R behavior [here](https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/decompose).

Reply:
Definitely will rename `type` 👍

Reply:
On second thought...I will follow [PEP8](https://pep8.org/#descriptive-naming-styles) and just add a single trailing underscore for python reserved keywords.

Reply:
Done!